src/flask/blueprints.py

```diff
@@ -376,9 +379,15 @@ def add_url_rule(
         if view_func and hasattr(view_func, "__name__") and "." in view_func.__name__:
             raise ValueError("'view_func' name may not contain a dot '.' character.")
 
-        self.record(lambda s: s.add_url_rule(rule, endpoint, view_func, **options))
+        self.record(
+            lambda s: s.add_url_rule(
+                rule, endpoint, view_func, provide_automatic_options, **options
```

Comment:
`provide_automatic_options` was a keyword argument before (via options) and will need to be here as well.

Reply:
I took a different approach of adding it as a positional argument, but passing it as a keyword argument seems much safer, given that I'm unfamiliar with this code and that `provide_automatic_options` doesn't seem to be well tested.

I just pushed a commit with the requested change. Should I squash the 3 commits for `add_url_rule()` together to reduce churn? And should I rebase against the branch, or do you not care? `CONTRIBUTING.rst` doesn't really go into the preferred Git workflow besides which branch to target.

Reply:
Please squash the commits - a cleaner history is nice. This is targeting the correct branch, and I should be able to merge without the rebase, but go ahead if you'd like.